src/components/Identity/index.js

```diff
@@ -38,39 +39,22 @@ const createIdentity = ({ config, logger, cookieJar }) => {
   const customerIds = {};
   let alreadyQueriedForIdSyncs = false;
 
-  const makeServerCall = payload => {
-    return lifecycle.onBeforeDataCollection(payload).then(() => {
-      return network.sendRequest(payload, payload.expectsResponse);
-    });
-  };
-
   const setCustomerIds = ids => {
     validateCustomerIds(ids);
     const event = createEvent(); // FIXME: We shouldn't need an event.
     event.mergeData({}); // FIXME: We shouldn't need an event.
     const payload = network.createPayload();
     payload.addEvent(event); // FIXME: We shouldn't need an event.
     assign(customerIds, ids);
-
-    const normalizedCustomerIds = normalizeCustomerIds(customerIds);
-
-    Object.keys(normalizedCustomerIds).forEach(idName => {
-      payload.addIdentity(idName, normalizedCustomerIds[idName]);
-    });
-    const customerIdsHash = crc32(
-      JSON.stringify(normalizedCustomerIds)
-    ).toString(36);
-    // TODO: add more tests around this piece
-    const customerIdChanged =
-      customerIdsHash !== cookieJar.get(CUSTOMER_ID_HASH);
-    payload.mergeMeta({ identityMap: { customerIdChanged } });
-    if (customerIdChanged) {
-      cookieJar.set(CUSTOMER_ID_HASH, customerIdsHash);
-    }
-    return lifecycle
-      .onBeforeEvent(event, {}, false) // FIXME: We shouldn't need an event.
-      .then(() => optIn.whenOptedIn())
-      .then(() => makeServerCall(payload));
+    syncDeclaredIds(
+      customerIds,
```

Comment:
How about we extract ALL the `setCustomerIds` logic in this module, instead of splitting the logic between here and `syncDeclaredIds`? Tell don't Ask right? logic in 2 places, even if it's only preparation can lead to bugs and confusion.